### PR TITLE
Updates for RC1

### PR DESCRIPTION
**Pull request by jhedstrom**
This gets all tests but one passing (and the one that isn't passing, isn't fatal erroring out of the test). Tests currently aren't runable due to RC1 changes.

### CODE REVIEW

**Comment by jhedstrom** [comment]
That last commit should get all the tests passing. Travis is still failing due to coding standards, which can probably be addressed in a follow-up.

**Comment by amitaibu** [comment]
Hi @jhedstrom 

> Travis is still failing due to coding standards

What do you mean by coding standards? AFAIK it's only running simple tests, not coder or stuff like that

File: src/Entity/Message.php [hunk 1]
```diff
@@ -302,21 +299,18 @@ public function getText($langcode = Language::LANGCODE_NOT_SPECIFIED, $options =
           $value = call_user_func_array($value['callback'], $value['arguments']);
         }
 
+        // @todo Replace some, or all, of this with core methods.
```
Comment (amitaibu) [inline src/Entity/Message.php]: I think this is pretty important, so we should tackle this here - what were you think of doing?

We should probably also mention `FormattableMarkup::placeholderFormat` here, which this function is now mimicking.

**Comment by jhedstrom** [comment]
It looks like [`phpcs` is run](https://travis-ci.org/Gizra/message/jobs/85558377), but doesn't cause a fail. The PHP 5.4 tests are failing (since it isn't supported by 8). I'll fix that (and look to add 5.6, and perhaps 7), and look into the core methods mentioned above later today.

**Comment by amitaibu** [comment]
I think it just runs the tests -- https://github.com/Gizra/message/blob/7.x-1.x/.travis.yml#L40

> and look into the core methods mentioned above later today.

Cool, basically it's a direct port of what `format_string` used to do in D7 - nothing too fancy.

**Comment by jhedstrom** [comment]
It didn't look like I could directly call the core method to replace what's going on here, so instead I changed that `@todo` to a `@see`.

Also, I updated `.travis.yml` to test php 5.5, 5.6, and 7.0. I marked 7.0 as allowed fail, but they are currently all passing!

**Comment by amitaibu** [comment]
Looking good to me, but haven't tested. Letting @RoySegall do the final merge :)

**Comment by RoySegall** [comment]
OK, It's look very good. 10x @jhedstrom.

I'll try to merge this during this week and also create a new release in d.o

**Comment by RoySegall** [comment]
@jhedstrom Iv'e checked the branch and when we have a clean installation of drupal 8 the message_example failed to install due to entity reference dependancy.  You can grant me a commit access or use a [path](https://gist.github.com/RoySegall/5ae775c0a3150f917116).

Another thing is that the messages displayed filtered HTML:
message__message____site-install

We can handle this in another PR which will also handle with removing calls to deprecated functions.

**Comment by jhedstrom** [comment]
Thanks @RoySegall. I applied the patch and pushed.

**Comment by RoySegall** [comment]
We still need to handle the html markup. I'll try to get to it.

**Comment by mccrodp** [comment]
Seems to be working as previous on RC4 now for this fork. I also ran the tests manually via UI and got:

```
An AJAX HTTP error occurred.
HTTP Result Code: 200
Debugging information follows.
Path: /batch?id=2&op=do_nojs&op=do
StatusText: OK
ResponseText: 
Drupal\message\Tests\MessageCron:
```

But the tests all seem to pass, so maybe it's not directly to do with this module.

> RESULTS
> 91 passes, 0 fails, 0 exceptions, 16 debug messages

@RoySegall I should have some time to help with the PRs left to get a release on DO. When do you think this can be merged? Thanks.

**Comment by mccrodp** [comment]
I made a start on [removing deprecated methods (patch)](https://gist.github.com/mccrodp/5fc2145d4b165e174488), included as a patch to current state of this PR as I'm not sure where is best to add this (i.e. - PR to a PR, etc?). All tests are passing.

I haven't looked at the html markup handling, but I can do if you point me towards recreating the problem.

**Comment by RoySegall** [comment]
@mccrodp you have another prs but this one strait lines with RC4 and maybe with 8.0.0. So I think we need to handle this PR first and then handle your PRs.

I think i might have some time this weekend to close the PR if some would not finish it first.

**Comment by mccrodp** [comment]
Yep, exactly @RoySegall, I'll merge upstream and continue with my PR when this gets in. Any details on what remains here to close this one? Perhaps I can give it a go if I get time this week.

**Comment by RoySegall** [comment]
The only problem I can see is the HTML tags - https://github.com/Gizra/message/pull/37#issuecomment-151284532

**Comment by mccrodp** [comment]
Where are you seeing this output in the UI so that I can recreate the problem? Is it in the Message Views?

**Comment by RoySegall** [comment]
If you'll enabled the `message_example` module and generate some node and comments using devel generate you'll get it.

**Comment by mccrodp** [comment]
Ok, thanks, hope to look at it this week if nobody else finishes it first.

**Comment by mccrodp** [comment]
@RoySegall, I created a [patch that wraps the markup into a render element](https://gist.github.com/mccrodp/8b3f933c5fd97ab8359f). However, I'm not sure this is the desired format in the view display: 
screenshot 2015-11-17 08 45 44

Did you want to do some kind of trim operation and escape certain markup, or is this ok?
I see there is a call to trim in the `MesageType::getText` already, but not in the `Message::getText`. It looks like it just removes carriage returns ("\n") though.

**Comment by RoySegall** [comment]
@mccrodp Your solution look elegant but I don't know if this is the proper solution. @jhedstrom What do you think about it?

Also, can you check what happens when the `trim` is removed?

**Comment by mccrodp** [comment]
@RoySegall, yes, I'm not so sure either, as the call to `MessageType::getText` also returns a Markup object then when called in `Message::getText`. So, I'm not sure if `MessageType::getText` is ever used / displayed directly, if it should return a Markup object or just a string.

Actually, there is a call to `implode` 1st in `MessageType::getText` which combines the message text items. `trim` is in fact only called to remove unwanted chars at the start or end, so this seems fine and it probably doesn't need to get called in `Message::getText` and is omitted currently.

I'll await feedback, it's a start anyway :)

**Comment by mccrodp** [comment]
@RoySegall, @jhedstrom This method may not be the complete solution if the way we access the partials in `MessageViewBuilder` is correct, but perhaps this is incorrect itself.

At the moment in `MessageViewBuilder` we have:

```
$partials = $entity->getType()->getText(NULL, array('text' => TRUE));
```

The 1st arg being `$langcode`, so we only load text in one language at a time. However, when accessing the data we seem to try to access it by language again.

```
$extra .= $partials[$langcode][$delta];
```

Perhaps this is incorrect as we don't return all partials indexed by language? i.e. Access like:

```
$extra .= $partials[$delta];
```

**Comment by mccrodp** [comment]
In order to continue work on Message UI upgrade I've decided to progress with a [forked version of this repo](https://github.com/mccrodp/message/tree/update-d8-stable) until this PR goes in. Made some progress that I hope to share (but also some regressions currently).

**Comment by jhedstrom** [comment]
Any chance of getting this merged? I just re-ran the tests against the latest `8.1.x` code and all tests are still green.

**Comment by mccrodp** [comment]
+1 This would be a great help getting this merged as-is and then follow up on the remaining issues in separate PRs. Thanks!

**Comment by RoySegall** [comment]
As i mentioned above, there is a lot of code between the work of all three of us so I merged it into a new PR and i mentioned what we missing for a D8 first stable release. 

So I would like to close this one and branch out from #39 branch.

**Comment by jhedstrom** [comment]
@RoySegall It doesn't matter to me which one is merged, I was reviewing all of them, and #39 has fails, while this one is all green, that's why I commented here. My main concern is that something gets merged so smaller PRs and issues can be more readily addressed. Thanks!

**Comment by RoySegall** [comment]
closing this one  infavor of #39 .

**Pull request closed without merging**